client/gutenberg/extensions/tiled-gallery/item.jsx

```diff
@@ -11,11 +14,11 @@ import { withSelect } from '@wordpress/data';
 
 class TiledGalleryImage extends Component {
 
-	UNSAFE_componentWillReceiveProps( { image, width, height } ) {
+	componentDidUpdate( { image, width, height } ) {
 		// very carefully set width & height attributes once only (to avoid recurse)!
-		if ( image && ! width && ! height && this.props.setAttributes ) {
+		if ( image && ! width && ! height && this.setAttributes ) {
```

Comment:
This doesn't look quite right especially with that comment. I'm quite sure it's not maintaining the previous behavior, but that looks buggy as well.

I think we want to update width and height whenever they change, does that sound right?

```jsx
componentDidUpdate( prevProps ) {
  if ( image && this.props.setAttributes ) {
    if ( prevProps.height !== this.props.height ) {
      this.props.setAttributes( { height: this.props.height } )
    }
    // …
  }
}
```

Also, why did `setAttributes` move out of props? All cases I see use `props.setAttributes` and there doesn't seem to be a method defined.

Reply:
Thanks! `props.setAttributes` is indeed correct :thumbsup:
I've been overlooking this section because I want to rewrite all the width/height stuff to be fluid, instead of fixed to 520px-width like it's now. I updated this `componentDidUpdate` in https://github.com/Automattic/wp-calypso/pull/27159/commits/4421529dbb38e75ae58a4029e87b56903c676977